Source/Phone/Call/DtmfQueue.swift

```diff
@@ -26,16 +26,14 @@ class DtmfQueue {
     
     var queue: [(event: String, completionHandler: CompletionHandler?)]
     let dispatchQueue: DispatchQueue
-    weak var call: Call!
     var correlationId: Int
     var waitingForResponse: Bool
-    
-    private let deviceUrl = DeviceService.sharedInstance.deviceUrl
-    
-    init(_ call: Call) {
+    private let callClient: CallClient
+
+    init(callClient: CallClient) {
```

Comment:
They're really not related. The Call was being passed in to get the participant url from it before calling a function that needed it. That's now being passed in as an argument. The CallClient was being newed up internally and using global data, now it's injected cleanly, and gets it's own dependencies injected as well.